proposals/buildfile.md

```diff
@@ -0,0 +1,186 @@
+# Jib CLI Buildfile Specification
+
+Specification for a YAML buildfile describing building a container image. This buildfile can be
+used by the jib-cli to generate a container. It is translated directly into a buildplan and
+passed to the builder.
+
+```yaml
+apiVersion: jib/v0alpha1
+kind: Config
+
+# "FROM" with detail for manifest lists or multiple architectures
```

Comment:
So what happens if the user does `jib build ./buildplan.yaml gcr.io/project/image` and `gcr.io/project/image` has never existed?  Do we always create a manifest list with a single image?  Do we only create a manifest list if there is > 1 platform specified?

What happens if `gcr.io/project/image` does exist and it's a manifest for a different platform (not a manifest list)?  Do we replace it?  Do we "promote" the existing image to a component of a manifest list?

Reply:
While this behavior will be determined by the underlying jib system, but yeah, it makes sense to at least document behavior here. I would think that each build with multiple platforms will create a new manifest list. It will not append.

Reply:
and all writes are overwrites

Reply:
added a section about multiple platforms.